text/0003-browser-js-built-in-metrics.md

```diff
@@ -0,0 +1,57 @@
+* Start Date: 2022-08-18
+* RFC Type: feature
+* RFC PR: https://github.com/getsentry/rfcs/pull/3
+* RFC Status: active
+
+# Expanding Built-In Performance Metrics for Browser JavaScript.
+
+# Summary
+
+This RFC details expanding list of built-in performance metrics for the browser JavaScript SDK, with additional data the Browser SDK captures. It propose adding two metrics that are already captured by the SDK: `connection.rtt` and `connection.downlink`, and three brand new metrics that are not yet captured, `device.memory`, `hardware.concurrency`, and `long_task.count`.
```

Comment:
Every new metric we extract increases the volume we need to ingest on the metrics pipeline and increases the storage size very quickly as suddenly all sdks start sending them for every transactions as soon as they are upgraded.

Is there a way we can phase such a rollout ? 
Right now we extract an average of 10 metrics per transaction. If we add 5 that represents 50%. Considering that javascript is one of the major sdks we have this will be a substantial increase.

Reply:
So 2 of these `connection.rtt`, and `connection.downlink` are already being sent by the SDK - but they are considered custom metrics instead of built-in metrics.

I agree though that we should consider phasing such a rollout, great suggestion. Ok so let's do this.

1) Convert `connection.rtt` and `connection.downlink` into built-in measurements. The value here is obv, and we are already collecting this data -> monitor the effect for 1 week
3) Add `hardware.concurrency` -> monitor the effect for 1 week
5) Add `device.memory` -> monitor the effect for 1 week

Based on some discussion with performance team and @k-fish we are dropping `long_task.count` from the proposal, it's not that valuable as we already collect the spans.

If we consider the effect not worth it, we can drop `device.memory`, but I feel we should not drop `hardware.concurrency`.

Reply:
Ok, 
also, something important to keep in mind.
If you add a standard tag to metrics or change the name of one (basically anything that is not custom), please register it on this file and statically assign it an ID.
https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/indexer/strings.py.

So the indexer will not assign a new ID to that tag key for every single organization and they all will share the same ID